profile/prune.go

```diff
@@ -130,7 +133,10 @@ func (p *Profile) PruneFrom(dropRx *regexp.Regexp) {
 				funcName := strings.TrimPrefix(fn.Name, ".")
 				// Account for unsimplified names -- trim starting from the first '('.
 				if index := strings.Index(funcName, "("); index > 0 {
-					funcName = funcName[:index]
+					// ... unless the thing in brackets is an anonymous namespace marker
+					if !strings.HasPrefix(funcName[index:], "(anonymous namespace)") {
```

Comment:
It feels that we will unexpectedly match a function like `Hello::(anonymous namespace)::World(const Foo::(anonymous namespace)::Test::Bar` with the `Foo::.*::Bar` pattern you're using. Can you please add such double-parens-pair test case to check whether it's so? If it is, I think it can be fixed by using LastIndex instead of Index above.